code/parachain/frame/pablo/src/lib.rs

```diff
@@ -105,13 +105,45 @@ pub mod pallet {
 		},
 	}
 
+	#[derive(
+		RuntimeDebug, Encode, Decode, MaxEncodedLen, CloneNoBound, PartialEq, Eq, TypeInfo,
+	)]
+	pub enum PoolConfigurationType {
```

Comment:
Is there no way to avoid this?

Reply:
The extra enum? probably yes, but it seemed the most clear/straightforward to me, and that makes it typed on Subsquid as well. We could use a simple string or `Vec<u8>` with the name but that might be more prone to errors. Any suggestion? Ben mentioned `core::mem::Discriminant`

Reply:
The extra enum? probably yes, but it seemed the most clear/straightforward to me, and that makes it typed on Subsquid as well. We could use a simple string or `Vec<u8>` with the name but that might be more prone to errors. Any suggestion? Ben mentioned `core::mem::Discriminant`

Reply:
I prefer an enum too. discrimants might change if we add enum variants to the top of an enum:

```
enum Foo {
    One // discriminant == 0
    Two // discriminant == 1
}

// adding a new type:
enum Foo {
    Three // discriminant == 0
    One // discriminant == 1
    Two // discriminant == 2
}

```

Reply:
@andresmechali you can use https://docs.rs/enum-kinds-macros/latest/enum_kinds_macros/ to generate the enum.

Reply:
Adding variants at the beginning of the enum will also break the scale encoding (unless we explicitly add the `#[codec(index = n)]` ofc, but still bad practice; any new variants should be added to the end of the enum). enum-kinds looks good though, and I'm not opposed to it, as long as it's only used internally in subsquid.

Reply:
Thanks @KaiserKarel  and @benluelo .
I think that to keep it simple and not having to maintain an extra enum I can live with just querying the type from storage when the pool is created and then query it internally on Subsquid when needed

Reply:
Also we avoid bloating the events with data that's only relevant for Subsquid, as Ben mentioned in other comment

Reply:
Given that pool creation is a low frequency event it's ok to query storage (though not a generally good practice for event sourcing systems because of the possibility of overloading upstream systems).